### PR TITLE
All Holodeck Bugs Must Die The Same Day Theyre Born or Your Money Back

### LINKED ISSUE #56635: Holodeck laser guns can make infinite lethal turrets.

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
Fixes #56635 
theres a good chance i should have just bit the bullet and forced yet another snowflakey if(flags_1 & HOLOGRAM_1) check that rejects laser guns being put into turrets if theyre a hologram but in my opinion the turrets should be makeable with it assuming they properly derez

## Why It's Good For The Game
athath put the eyes reaction on the bug report, making this so he cant farm laser guns via the 70% turret drop rate strategy hes no doubt already made.

## Changelog
:cl:
fix: turrets made via the holodeck laser guns now stop working when the guns get deleted
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
image

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -188,6 +193,10 @@ DEFINE_BITFIELD(turret_flags, list(
 	update_icon()
 	return gun_properties
 
+///destroys reference to stored_gun to prevent hard deletions
+/obj/machinery/porta_turret/proc/null_gun()
```
Comment (Jared-Fogle) [inline code/game/machinery/porta_turret/portable_turret.dm:197]: Needs SIGNAL_HANDLER

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 2]
```diff
@@ -404,7 +413,7 @@ DEFINE_BITFIELD(turret_flags, list(
 				cover = new /obj/machinery/porta_turret_cover(loc)	//if the turret has no cover and is anchored, give it a cover
 				cover.parent_turret = src	//assign the cover its parent_turret, which would be this (src)
 
-	if(!on || (machine_stat & (NOPOWER|BROKEN)) || manual_control)
+	if(!on || (machine_stat & (NOPOWER|BROKEN)) || manual_control || (!stored_gun && uses_stored))
```
Comment (Jared-Fogle) [inline code/game/machinery/porta_turret/portable_turret.dm:416]: This is getting confusing, split this into multiple if statements.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 2 above)
Comment (Kylerace) [reply code/game/machinery/porta_turret/portable_turret.dm]: it is now three

File: code/game/machinery/porta_turret/portable_turret.dm (hunk 1 above)
Comment (Kylerace) [reply code/game/machinery/porta_turret/portable_turret.dm:197]: oops, fixed

**Comment by Jared-Fogle** [comment]
youre going to get banned one day ath

**Pull request merged**